### PR TITLE
[WIP] CacheRepository unit tests

### PR DESCRIPTION
**Pull request by joselufo**
Pull request goal is to cover by unit tests the `CacheRepository` class

### CODE REVIEW

File: harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/repository/CacheRepository.kt [hunk 1]
```diff
@@ -30,16 +30,17 @@ class CacheRepository<V>(
     return when (operation) {
       is DefaultOperation -> get(query, CacheSyncOperation())
       is MainOperation -> getMain.get(query)
-      is CacheOperation -> getCache.get(query).let {
-        try {
-          if (!validator.isValid(it)) {
+      is CacheOperation -> {
+        val cacheValue = getCache.get(query)
+        return try {
```
Comment (franmontiel) [inline harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/repository/CacheRepository.kt:35]: Just noticed that this "try" should be wrapping also the getChage.get(query) as it could throw an exception.

This is not a huge problem but it is worth changing it for two reasons:
- We are not receiving that possible exception on the fallback lambda and we should
- CacheSyncOperation wrap the cache call inside the try/catch (so this one should behave the same)

It is not something related to your changes so I am approving the PR. We must change this on this PR or in the following one fixing the CacheSyncOperation fallback

File: harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/repository/CacheRepository.kt [hunk 2]
```diff
@@ -85,21 +86,24 @@ class CacheRepository<V>(
     return when (operation) {
       is DefaultOperation -> getAll(query, CacheSyncOperation())
       is MainOperation -> getMain.getAll(query)
-      is CacheOperation -> getCache.getAll(query).map {
-        try {
-          if (!validator.isValid(it)) {
+      is CacheOperation -> {
+        val cacheValues = getCache.getAll(query)
```
Comment (franmontiel) [inline harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/repository/CacheRepository.kt:90]: getCache.getAll should be inside the try/catch. Same issue mentioned on get() method

File: harmony-kotlin/src/commonTest/kotlin/com/harmony/kotlin/data/repository/CacheRepositoryTests.kt [hunk 3]
```diff
@@ -0,0 +1,463 @@
...
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+
+@ExperimentalCoroutinesApi
+class CacheRepositoryTests {
+
+  //region get() - tests
+
+  @Test
+  fun `should retrieves the value from the main datasource when MainOperation is provided when get() is called`() = runTest {
```
Comment (franmontiel) [inline harmony-kotlin/src/commonTest/kotlin/com/harmony/kotlin/data/repository/CacheRepositoryTests.kt:39]: Picky suggestion, no need to change it: To understand the test better I would avoid those two 'whens'

Maybe rephrase from this:
- should retrieves the value from the main datasource **when** MainOperation is provided **when** get() is called
to something like:
- get() should retrieve the value from the main datasource when MainOperation is provided

The same happens with the rest of the tests

**Review by franmontiel — APPROVED** [review]
Approved. Check my comments and decide whether or not it is worth changing them in this PR. Two things noticed:
- Previous issue with get()/getAll() and cacheOperation not wrapping the cache call into the try/catch. If not addressed here let me know and we should address it this week.
- A suggestion on naming

File: harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/repository/CacheRepository.kt (hunk 1 above)
Comment (franmontiel) [reply harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/repository/CacheRepository.kt:35]: Fix applied to #60

File: harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/repository/CacheRepository.kt (hunk 2 above)
Comment (franmontiel) [reply harmony-kotlin/src/commonMain/kotlin/com.harmony.kotlin/data/repository/CacheRepository.kt:90]: Fix applied to #60

File: harmony-kotlin/src/commonTest/kotlin/com/harmony/kotlin/data/repository/CacheRepositoryTests.kt (hunk 3 above)
Comment (franmontiel) [reply harmony-kotlin/src/commonTest/kotlin/com/harmony/kotlin/data/repository/CacheRepositoryTests.kt:39]: Some changes applied to #60

**Review by jfilowk — APPROVED** [review]
